packages/ra-ui-materialui/src/button/SortButton.tsx

```diff
@@ -43,6 +43,7 @@ const SortButton: FC<{ fields: string[]; label?: string }> = ({
     fields,
     label = 'ra.sort.sort_by',
 }) => {
+    const { resource } = useListContext();
```

Comment:
no need to call 2 contexts, please move currentSort and setSort here and remove useListSortContext

Reply:
Done